new lexer : SINEX file format

freinquin opened the pull request:
The SINEX format is a block-oriented file format used in science, and especially space geodesy. It was created about 30 years ago (https://www.iers.org/SharedDocs/Publikationen/EN/IERS/Documents/ac/sinex/sinex_v202_pdf.pdf). The files are usually created by programs (just like XML files usually are) but it would be very useful to navigate through those files via a SINEX-enabled text editor.
I have a working implementation for geany :
<img width="885" height="675" alt="image" src="https://github.com/user-attachments/assets/111361da-d2c2-46f0-b770-fdbc2c9248b9" />
It is my understanding that the code must be available in official releases of ctags & scintilla/lexilla before submitting a pull request on the geany repo. The ctags part was included in July, this is the lexilla part.

freinquin commented on lexers/LexSINEX.cxx:
```diff
@@ -0,0 +1,314 @@
+
+using namespace Lexilla;
+
+// States when parsing a real number
```
Thanks for the feedback. Done.

nyamatongwe commented:
`SCE_SINEX_STRING` is defined but never used.

There's a warning from cppcheck for lengthLine. If it can be less than 0 then passing as unsigned is dangerous.
~~~
lexilla\lexers\LexSINEX.cxx:162:17: warning: Checking if unsigned expression 'lengthLine' is less than zero. [unsignedLessThanZero]
 if (lengthLine <= 0)
                ^
~~~

freinquin commented:
> `SCE_SINEX_STRING` is defined but never used.

You're right. Now removed.
> 
> There's a warning from cppcheck for lengthLine. If it can be less than 0 then passing as unsigned is dangerous.
> 
Can't happen. The passed argument has type "size_t" (unsigned something). Anyway, my test is an old habit of trying and catch all possible values and more. Changed to a mere "==".
